### ISSUE TITLE
Allow profiling of queries via the API

### ISSUE THREAD

**Issue by oren** [cayleygraph/cayley#328]
It is currently possible to profile queries via the go `Describe()` function but it will nice to have it available from the API. 

Here is an example of Neo4J `profile` command. I think we should first expose it to the console and maybe later on to the web interface.
![profile-neo](http://i.stack.imgur.com/8RzHZ.png)

**Comment by robertmeta** [comment]
Yep, interesting, moved to feature request forum for now: https://discourse.cayley.io/t/allow-profiling-of-queries-via-the-api/229

**Comment by dennwc** [comment]
With Shapes in place it's becomes even more relevant. Use #670 to show query plan and use contexts that we pass around to enable profiling hooks to get live query stats.
